python/GafferSceneUI/ShaderUI.py

```diff
@@ -621,6 +621,131 @@ def __parameterValues( self ) :
 
 		return None
 
+# A path filter that keeps a path if it, any of its ancestors or any of its descendants have a
+# property `propertyName` that matches `patterns`.
+
+class _PathMatcherPathFilter( Gaffer.PathFilter ) :
+
+	def __init__( self, patterns, rootPath = None, propertyName = "name", userData = {} ) :
+
+		Gaffer.PathFilter.__init__( self, userData )
+
+		self.__patterns = patterns
+		self.__rootPath = rootPath
+		self.__propertyName = propertyName
+
+		self.__pathMatcherDirty = True
+		self.__pathMatcher = IECore.PathMatcher()
+
+		self.setRootPath( rootPath )
+
+	def setMatchPatterns( self, patterns ) :
+
+		if self.__patterns == patterns :
+			return
+
+		self.__patterns = patterns
+		self.__pathMatcherDirty = True
+		self.changedSignal()( self )
+
+	def getMatchPatterns( self ) :
+
+		return self.__patterns
+
+	def setRootPath( self, rootPath ) :
+
+		if rootPath is None :
+			self.__rootPathChangedConnection = None
+			self.__rootPath = rootPath
+			self.__paths = []
+			self.__pathMatcherDirty = True
+			return
+
+		if self.__rootPath == rootPath :
+			return
+
+		self.__rootPath = rootPath
+		self.__updatePaths()
+
+		self.__rootPathChangedConnection = self.__rootPath.pathChangedSignal().connect(
+			Gaffer.WeakMethod( self.__rootPathChanged ),
+			scoped = True
+		)
+		self.__pathMatcherDirty = True
+
+	def getRootPath( self ) :
+
+		return self.__rootPath
+
+	def setPropertyName( self, propertyName ) :
+
+		if( self.__propertyName == propertyName ) :
+			return
+
+		self.__propertyName = propertyName
+		self.__pathMatcherDirty = True
+		self.changedSignal()( self )
+
+	def getPropertyName( self ) :
+
+		return self.__propertyName
+
+	def _filter( self, paths, canceller ) :
+
+		if len( paths ) == 0 :
+			return []
+
+		self.__updatePathMatcher()
+
+		if self.__pathMatcher.isEmpty() :
+			return []
+
+		result = [ p for p in paths if self.__pathMatcher.match( str( p ) ) ]
+
+		return result
+
+	def __updatePathMatcher( self ) :
+
+		if not self.__pathMatcherDirty or len( self.__patterns ) == 0 or self.__propertyName == "" :
```

Comment:
Won't a call to `setPropertyName( "" )` or `setMatchPatterns( [] )` fail to update the path matcher?

Reply:
Right! Fixed in https://github.com/GafferHQ/gaffer/pull/5352/commits/100525f5b08bf7b40e5de70e813a9219748c31be